mods/dye/init.lua

```diff
@@ -20,13 +20,104 @@ dye.dyes = {
 	pink		= { name = "Pink", html = "#FF7FFF" },
 }
 
--- Define items
+----
+-- Helper functions.
+----
+
+--[[
+	Convert an html color in the format of #RRGGBB into red, green, and blue values on the range of [0..255].
+]]
+function hex2rgb(hex)
+    hex = hex:gsub("#","")
+    return tonumber("0x"..hex:sub(1,2)), tonumber("0x"..hex:sub(3,4)), tonumber("0x"..hex:sub(5,6))
+end
+
+function rgb2hex(r, g, b)
+	return string.format("#%02X%02X%02X", r, g, b)
+end
+
+
+--[[
+	Convert red, green, and blue values from the range of [0..255]
+	into hue, saturation, and value/lightness values on the range of...?
+]]
+function rgbToHsv(red, green, blue)
+	local r = red / 255
+	local g = green / 255
+	local b = blue / 255
+	local max = math.max(r, g, b)
+	local min = math.min(r, g, b)
+	local h, s, v = max, max, max
+
+	local d = max - min;
+	if max == 0 then
+		s = 0
```

Comment:
That's what I get for cutting-and-pasting code.  I can standardize this as well.